streams/src/main/java/org/apache/kafka/streams/state/KeyValueStore.java

```diff
@@ -51,6 +51,16 @@
     abstract public void put(K key, V value);
 
     /**
+     * Update the value associated with this key, unless a value
+     * already is associated with the key
```

Comment:
Since a state store is only accessed by a single thread, it should be atomic by itself. But we can make this statement in the comment.